### DISCUSSION TITLE
Inverse filter in the layout view

### DISCUSSION THREAD

**Discussion by cristobalbahe** [directus/directus#7931] [Feature Requests]
Hello everyone,

I was wondering if there was some way to inverse the results from a filtered list in the collections view. I will explain my case:

I have many galleries that have many exhibitions. Some of them are currently open since all exhibitions have a field with opening and closing date.

If I filter my list of galleries to show me the exhibitions that open and close time are greater than today I get all galleries with current exhibitions. But since all past exhibitions are still linked to the gallery, I cannot filter this list with filters like "less than" so that I can see only those galleries that **do not** have an open exhibition.

So I thought that maybe it would be useful to filter the list so that it shows the items that don't have the condition so that I can do the next: 

Filter a list with the items that do not satisfy X condition.

Maybe there is some way to do this and I'm being completely dumb, but I would love to hear from someone that maybe has a solution that I can implement and if not, just lay it on the table :)

Cheers everyone and congrats again on making such a wonderful app.

**Comment by cristobalbahe** [comment]
Another way of doing this, maybe, would be applying the condition to all of the items. I mean, if all exhibitions close date is less than today, then it would give the same result

**Comment by rijkvanzanten** [comment]
What do you mean by the following?:

>  I cannot filter this list with filters like "less than" so that I can see only those galleries that do not have an open exhibition.

There is a "is less than" filter operator, and we have a negated version for every other operator (eg "is null" has a "is NOT null") as well, so I believe that should cover this use case 🤔

**Reply by cristobalbahe** [reply]
Since i am filtering a o2m relationship (a gallery has many exhibitions) and all exhibitions meet the the condition "opening date < now" except the current one, the less than filter returns all galleries, instead of only those galleries that only have past exhibitions and no current exhibition, which is what I would need. 

So if I could apply "less than" this way: 

> (All items opening date < now) is true
> (Some items opening date > now) is false

I would get what i meant. 

On the contrary I can easily get the galleries with current exhibitions with "opening date > now" which excludes all those that do not have current exhibition. If I could inverse the selection resulting from this filter i would also get the same result. 

I don't know if this makes it any clearer, to be honest 😅

**Reply by rijkvanzanten** [reply]
Took a couple reads, but I think I understand 😁 

* You have "Galleries" and "Exhibitions" each gallery can have one or more exhibitions. 
* An exhibition has an open and close date.
* You're trying to get all "Galleries" that have an open exhibition

If that's correct, you should be able to get what you're looking for with two filters:

```
GET /items/galleries
  ?filter[exhibitions][open_date][_lte]=$NOW
  &filter[exhibitions][close_date][_gte]=$NOW
```

**Reply by rijkvanzanten** [reply]
Oh hold up, you increase the stakes another level:

* You're trying to get all "Galleries" that DO NOT have any open exhibitions

I believe you can achieve that by relying on the "_or" operator. Galleries that don't _currently_ have any  open exhibitions basically means the same as "galleries with exhibitions that are already closed, or have yet to open", which can be achieved with:

```json
{
  "_or": [
    {
      "exhibitions": {
        "close_date": {
          "_lte": "$NOW"
        }
      }
    },
    {
      "exhibitions": {
        "open_date": {
          "_gte": "$NOW"
        }
      }
    }
  ]
}
```

**Reply by cristobalbahe** [reply]
Hello again! Filtering that way doesn't get the results of galleries that don't have open exhibitions. Here is an example:

```javascript
galleries: [
  {
    name: "gallery1",
    exhibitions: [
      {
        //This is a current exhibition
        name: "Exhibition 1",
        opening: "2021-09-01",
        closing: "2021-09-30",
      },
      {
        //This is a past exhibition
        name: "Exhibition 2",
        opening: "2021-08-01",
        closing: "2021-08-30",
      },
    ],
  },
];
```

If I do 

```javascript
galleries ( 
    filter: {
        expositions: {
            closing: {
                _lte: "$NOW"
            }
        }
    }
)
```

The exhibition with name "Exhibition 2" meets the condition, since it has already closed, so the gallery with name "gallery1" will be returned, although it has a current exhibition ("Exhibition 1")
(Thanks again for your time!)

**Reply by rijkvanzanten** [reply]
Ahhh yeah good one. That relational filtering structure works on a "If one of the items has X" approach instead of a "If _all_ nested times have X" currently. Ideally, we add in some way to specify that with "_each" and "_some" for example, but that's currently not possible 🤔

**Reply by cristobalbahe** [reply]
Ahhh, I see! Okay, thank you so much!

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
